modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/IgniteMdRowCount.java

```diff
@@ -47,6 +48,63 @@
         return joinRowCount(mq, rel);
     }
 
+    /** */
+    public Double getRowCount(IgniteCorrelatedNestedLoopJoin rel, RelMetadataQuery mq) {
+        if (!rel.getJoinType().projectsRight()) {
+            // Create a RexNode representing the selectivity of the
+            // semijoin filter and pass it to getSelectivity
+            RexNode semiJoinSelectivity =
+                RelMdUtil.makeSemiJoinSelectivityRexNode(mq, rel);
+
+            return multiply(mq.getSelectivity(rel.getLeft(), semiJoinSelectivity),
+                mq.getRowCount(rel.getLeft()));
+        }
+
+        // Row count estimates of 0 will be rounded up to 1.
+        // So, use maxRowCount where the product is very small.
+        final Double left = mq.getRowCount(rel.getLeft());
+        final Double right = mq.getRowCount(rel.getRight());
+
+        if (left == null || right == null)
+            return null;
+
+        if (left <= 1D || right <= 1D) {
```

Comment:
replace 10 with constant.

Reply:
it is 1D (one of type Double). To be honest I don't sure we need a constant for this